bec_alerts/triggers.py

```diff
@@ -0,0 +1,169 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+"""
+This module contains alert trigger definitions that describe when alerts
+should be sent to users. To subscribe to an alert, you must add it to
+this file as a new subclass of the Trigger class, or add your email to
+the list of emails for an existing trigger subclass.
+"""
+import logging
+from datetime import timedelta
+
+from django.template.loader import render_to_string
+from django.utils import timezone
+from django.utils.functional import cached_property
+
+from bec_alerts.models import IssueBucket, User, UserIssue
+
+
+class Trigger:
+    """
+    Parent class for alert triggers. Subclasses should override methods
+    in this class to customize the behavior of their alert.
+    """
+    #: name is ideally unique, and is included by default in the subject
+    name = 'Error'
+
+    # List of emails to alert when triggered
+    emails = []
+
+    #: Path to the template file to use for alerts sent by this trigger.
+    #: Paths are relative to the jinja2 directory and use Jinja2 for
+    #: templating.
+    template = 'basic_alert.txt'
```

Comment:
The templates seem "short-ish". I think I would have done them in a triple-quoted string rather than a separate file and done each trigger as a separate Python module.

Reply:
I'm not suggesting you need to change anything--just thinking out loud.

Reply:
For emails, whitespace is significant (unless you're doing, like, html emails), and tracking the state of whitespace in an indented docstring is kinda difficult and finnicky. I usually lean towards standalone templates in these cases, and Jinja is especially helpful with the `-` modifier for collapsing whitespace.